### PR TITLE
CASMINST-3853 - Move kernel install and lock to common layer

### PR DESCRIPTION
**Pull request by manderson-hpe**
#### Summary and Scope
This change moves the kernel installation and locking to the common layer. 

This change enables kernel changes without rebuilding the base SLES layer, and further decouples the layer builds from the base installation. The change also removes duplicate kernel locking code, and a rogue installation and check of `jq`. `jq` has been moved to the base installation layer as it is required across the installation process for the rpm tools.

### CODE REVIEW

File: boxes/sles15-base/http/autoinst.template.xml [hunk 1]
```diff
@@ -114,6 +114,7 @@
       <package>squashfs</package>
       <package>sudo</package>
       <package>tar</package>
+      <package>jq</package>
```
Comment (rustydb) [inline boxes/sles15-base/http/autoinst.template.xml:117]: Alphabetize

**Review by rustydb — CHANGES_REQUESTED** [review]

File: boxes/sles15-base/http/autoinst.template.xml (hunk 1 above)
Comment (manderson-hpe) [reply boxes/sles15-base/http/autoinst.template.xml:117]: I apologize for this oversight.

File: boxes/sles15-base/variables.pkr.hcl [hunk 2]
```diff
@@ -69,11 +69,6 @@ variable "artifact_version" {
   default = "none"
 }
 
-variable "kernel_version" {
-  type    = string
-  default = "5.3.18-59.34.1"
```
Comment (heemstra) [inline boxes/sles15-base/variables.pkr.hcl:74]: Where/how is this getting set now?

Comment (rustydb) [reply boxes/sles15-base/variables.pkr.hcl:74]: I (maybe) assumed this was a default for local builds to work, but I'm now more curious too. I suppose CSM-rpms would override this when it does `zypper install` before it locks the kernel?

Comment (heemstra) [reply boxes/sles15-base/variables.pkr.hcl:74]: Looks like this is removing the duplicate code from sles15-base.  The same thing still exists in common.

**Pull request closed without merging**